### ISSUE TITLE
API: Functions for calling Methods which are not derived by Method_Solver class

### ISSUE THREAD

**Issue by smavros** [spirit-code/spirit#368]
In __Spirit v1.7.0__ the API function provides a family of functions ```Simulation_*( ... )``` for acting on the state with a specific method like

```c++
Simulation_PlayPause( State *, const char * c_method_type, const char * c_solver_type, int n_iterations, int n_iterations_log, int idx_image, int idx_chain );
```
That family of functions require the method type (via ```c_method_type``` variable) and the type of the solver used by the method (```c_solver_type```). That implies that every method uses a solver (it is a child class of ```Method_Solver```). 

__Problem:__ That does not hold true for every ```Method``` since we have a method which does not use any solver (MC is child classes of ```Method```) and in the future we will probably add more. When it comes to MC a work around with the given API is to call the ```Simulation_*()``` functions with an arbitrary ```c_solver_type``` since a solver it is not used at all. Of course, this situation may not be clear for the user and creates confusion.

__Proposal:__ For every ```Method``` that derives form ```class Method``` we should probably add a corresponding API function e.g.

```c++
Simulation_MC( State *, int n_iterations, int n_iterations_log, int idx_image, int idx_chain );
```

**Comment by GPMueller** [comment]
Done with 02682726c931b2b774044e99b1c6b2085e519752 on `feature-improved-api`.

**Comment by GPMueller** [comment]
Merged into `develop` with eb841c5bd39941d4c437895845969c00ff224f10.
